esmvalcore/_recipe.py

```diff
@@ -1010,6 +1010,10 @@ def _initialize_variables(self, raw_variable, raw_datasets):
         raw_variable = deepcopy(raw_variable)
         datasets = self._initialize_datasets(
             raw_datasets + raw_variable.pop('additional_datasets', []))
+        if not datasets:
+            raise RecipeError(f"You have not specified any dataset "
+                              f"or additional_dataset groups "
+                              f"for variable %s Exiting." % raw_variable)
```

Comment:
```suggestion
            raise RecipeError("You have not specified any dataset "
                              "or additional_dataset groups "
                              f"for variable {raw_variable} Exiting.")
```
Why the sudden change to `%` style string formatting?